### PR TITLE
[WIP] Bugfix strict param types master merged

### PR DESCRIPTION
**Pull request by dmytro-dymarchuk**

### CODE REVIEW

**Comment by dereuromark** [comment]
Any such change should now target the new branch https://github.com/propelorm/Propel2/pull/1843 as per suite nonsplit composer.json declared

File: src/Propel/Runtime/Formatter/ArrayFormatter.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function getCollectionClassName(): ?string
      *
      * @return array|null
      */
-    public function formatOne(?DataFetcherInterface $dataFetcher = null): ?array
+    public function formatOne(?DataFetcherInterface $dataFetcher = null)
```
Comment (dereuromark) [inline src/Propel/Runtime/Formatter/ArrayFormatter.php:91]: Why is this needed?

Comment (dmytro-dymarchuk) [reply src/Propel/Runtime/Formatter/ArrayFormatter.php:91]: We have two child classes that return transfers.
`\Spryker\Zed\Kernel\Persistence\Repository\TransferObjectFormatter::formatOne()`
`\Spryker\Zed\Synchronization\Persistence\Propel\Formatter\SynchronizationDataTransferObjectFormatter::formatOne()`

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php [hunk 2]
```diff
@@ -7182,7 +7182,7 @@ public function clear()
         \$this->resetModified();
         \$this->setNew(true);
         \$this->setDeleted(false);
-
+
```
Comment (dereuromark) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php:7185]: invalid change?

File: src/Propel/Runtime/Connection/ConnectionManagerPrimaryReplica.php [hunk 3]
```diff
@@ -31,7 +31,7 @@ class ConnectionManagerPrimaryReplica implements ConnectionManagerInterface
     protected $writeConnection;
 
     /**
-     * @var array
+     * @var array|null
```
Comment (dereuromark) [inline src/Propel/Runtime/Connection/ConnectionManagerPrimaryReplica.php:34]: not needed anymore
It is never null

File: src/Propel/Runtime/Formatter/ArrayFormatter.php (hunk 1 above)
Comment (dereuromark) [reply src/Propel/Runtime/Formatter/ArrayFormatter.php:91]: If there is no other way, please fix the docblock to

    @return array|mixed|null
